hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/block/transformer/TokenAirdropTransformer.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (C) 2025 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.mirror.importer.downloader.block.transformer;
+
+import static com.hedera.hapi.block.stream.output.protoc.StateIdentifier.STATE_ID_PENDING_AIRDROPS;
+
+import com.hedera.hapi.block.stream.output.protoc.StateChanges;
+import com.hedera.mirror.common.domain.transaction.BlockItem;
+import com.hedera.mirror.common.domain.transaction.TransactionType;
+import com.hederahashgraph.api.proto.java.AccountAmount;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.NftID;
+import com.hederahashgraph.api.proto.java.NftTransfer;
+import com.hederahashgraph.api.proto.java.PendingAirdropId;
+import com.hederahashgraph.api.proto.java.PendingAirdropId.TokenReferenceCase;
+import com.hederahashgraph.api.proto.java.PendingAirdropRecord;
+import com.hederahashgraph.api.proto.java.TokenID;
+import com.hederahashgraph.api.proto.java.TokenTransferList;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+import com.hederahashgraph.api.proto.java.TransactionRecord;
+import jakarta.inject.Named;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Named
+final class TokenAirdropTransformer extends AbstractBlockItemTransformer {
+
+    @Override
+    protected void updateTransactionRecord(
+            BlockItem blockItem, TransactionBody transactionBody, TransactionRecord.Builder transactionRecordBuilder) {
+        if (!blockItem.successful() || !transactionBody.hasTokenAirdrop()) {
```

Comment:
the token airdrop transformer should only be called for token airdrop transactions. violating that contract is a bigger problem.

```suggestion
        if (!blockItem.successful()) {
```

Reply:
Thanks, I have removed that clause.